api/v1alpha1/rollout_types.go

```diff
@@ -121,6 +121,8 @@ type CanaryStrategy struct {
 	// only support for canary deployment
 	// +optional
 	PatchPodTemplateMetadata *PatchPodTemplateMetadata `json:"patchPodTemplateMetadata,omitempty"`
+	// canary service will not be generated if DisableGenerateCanaryService is true
+	DisableGenerateCanaryService bool `json:"disableGenerateCanaryService,omitempty"`
```

Comment:
v1alpha1 is no longer considered to support this field, in addition, all the field capabilities of this patch are not considered to be supported in v1alpha1.